presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MetadataQueryOptimizer.java

```diff
@@ -181,7 +187,14 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Void> cont
 
             if (isReducible(node, inputs)) {
                 // Fold min/max aggregations to a constant value
-                return reduce(node, inputs, columns, context, discretePredicates);
+                return reduce(node, inputs, columns, context, discretePredicates, tableScan.getTable());
+            }
+
+            // Disable the rewrite if the metadata of the any involved partition cannot be trusted.
+            for (TupleDomain<ColumnHandle> tupleDomain : discretePredicates.getPredicates()) {
```

Comment:
I added a separate check in reduce as an optimization. For the reduce path, we only need to check the result partition instead of all involved partitions.